services/dedup/scylla/scylla.go

```diff
@@ -33,6 +33,60 @@ func (d *ScyllaDB) Close() {
 	d.scylla.Close()
 }
 
+func (d *ScyllaDB) GetBatch(kvs []types.KeyValue) (map[types.KeyValue]bool, map[types.KeyValue]int64, error) {
+	defer d.stat.NewTaggedStat("dedup_get_batch_duration_seconds", stats.TimerType, stats.Tags{"mode": "scylla"}).RecordDuration()()
+	// Prepare a map to store results for each job (true = accept, false = reject)
+	results := make(map[types.KeyValue]bool)
+	sizes := make(map[types.KeyValue]int64)
+
+	// Group jobs by workspaceID for batch querying
+	workspaceJobsMap := make(map[string][]types.KeyValue)
+	d.cacheMu.Lock()
+
+	for _, kv := range kvs {
+		if previous, found := d.cache[kv.Key]; found {
+			results[kv] = false
+			sizes[kv] = previous.Value
+			continue
+		}
+		d.cache[kv.Key] = kv
+		results[kv] = true
+		sizes[kv] = kv.Value
+		workspaceJobsMap[kv.WorkspaceID] = append(workspaceJobsMap[kv.WorkspaceID], kv)
+	}
+	d.cacheMu.Unlock()
+
+	// Loop over each workspace and query all messageIDs for that workspace
+	for workspaceID, workspaceJobs := range workspaceJobsMap {
+		var messageIDs []string
+		for _, job := range workspaceJobs {
+			messageIDs = append(messageIDs, job.Key)
+		}
+
+		// Query to get all jobIDs for the given workspaceID and messageIDs
+		query := fmt.Sprintf("SELECT id, size FROM %s.%q WHERE workspaceID = ? AND id IN ?", d.keyspace, d.tableName)
```

Comment:
instead of querying for every batch can we do something like this? 
```
SELECT id, size
FROM ks.table 
WHERE (workspace_id = x AND message_id IN (list_of_message_ids_x)) 
   OR (workspace_id = y AND message_id IN (list_of_message_ids_y));
```

Reply:
This will not reduce qps iter will only do a paginated query. Not exactly sure on how iter is implemented for scylla, but I thought it would be similar to that of postgres.

Reply:
discussed offline, this way we would be querying multiple partitions which is not recommended by scylladb